Source/WPEFramework/ControllerJsonRpc.cpp

```diff
@@ -547,6 +559,202 @@ namespace Plugin {
         return result;
     }
 
+    // Property: version of the controller
+    // Return codes:
+    //  - ERROR_NONE: Success
+    //  - ERROR_GENERAL: Failed to get the version
+    uint32_t Controller::get_version(Core::JSON::String& response) const
+    {
+        uint32_t result = Core::ERROR_NONE;
+   
+        ASSERT(_pluginServer != nullptr);
+        
+        if (_pluginServer ==nullptr)  {
+            result = Core::ERROR_GENERAL;
+        }
+
+        response = _pluginServer->_config.Version();
+    
+        return  result ;
+        
+    }
+
+    // Property: version of the controller
+    // Return codes:
+    //  - ERROR_NONE: Success
+    //  - ERROR_GENERAL: Failed to set the version
+    uint32_t Controller::set_version(const Core::JSON::String& params)
+    {
+         uint32_t result = Core::ERROR_NONE;
+        
+        ASSERT(_pluginServer != nullptr);
+        
+        if (_pluginServer ==nullptr)  {
```

Comment:
In principle the ASSERT should suffice but the way the if is now I predict it will never return the error but crash Thunder :)

Reply:
Assert is only in debug mode right ? indeed Thunder will crash in that case. I corrected the logic in all functions.